### ISSUE TITLE
Bring back TypeScript support for 3.x

### ISSUE THREAD

**Issue by Znegl** [Cropster/ember-l10n#54]
After updating to version 3.0.0 running `ember l10n:extract` results in the following error:

```
Extracting app/adapters/application.ts >>> messages.pot...
Line 17: Unexpected identifier
```

Line 17 is where the typescript declaration sits:
```
declare module 'ember-data' {
  interface AdapterRegistry {
    'application': Application;
  }
}
```

Is there any way to make it ignore the declarations?

**Comment by mydea** [comment]
Hmm, sorry to hear that. I was not aware that esprima - what we now use to parse .js files - does not support TypeScript.

Right now, we don't really have the resources to implement this - so you probably need to stay on 2.x for now.

However, we'd love to accept a PR for that! Feel free to ask if you need any guidance. Roughly speaking, this is what would need to be done:

- [ ] Create a new file `lib/utils/commands/utils/parse-ts.js` based on https://github.com/Cropster/ember-l10n/blob/master/lib/commands/utils/parse-js.js
- [ ] Look for a good TS AST parser - basically an equivalent to http://esprima.org/. Replace the esprima logic from `parse-js` with the AST logic of the TS parser.
- [ ] In the file `/lib/commands/extract.js`, adapt the method `_extractFromJS()` to call `parseJsFile()` only on files ending with `.js`, and call the new `parseTsFile()` afterwards for files ending with `.ts`. You can use `let jsFiles = this._getFiles(options, 'js');` and `let tsFiles = this._getFiles(options, 'ts');` to get the appropriate files.

The whole method would look something like this:

```js
_extractFromJS(options) {
    let jsFiles = this._getFiles(options, 'js');
    let tsFiles = this._getFiles(options, 'ts');

    let outputFile = this.potFile;
    let potFile = this._getFileName(outputFile);

    jsFiles.forEach(file => {
      if (this._shouldSkipFile(file)) {
        this.ui.writeLine(chalk.yellow(`Skipping ${file}...`));
        return;
      }
      
      this.ui.writeLine(chalk.white(`Extracting ${file} >>> ${potFile}...`));

      parseJsFile(file, options, this.messageGettextItems);
    });

    tsFiles.forEach(file => {
      if (this._shouldSkipFile(file)) {
        this.ui.writeLine(chalk.yellow(`Skipping ${file}...`));
        return;
      }
      
      this.ui.writeLine(chalk.white(`Extracting ${file} >>> ${potFile}...`));

      parseTsFile(file, options, this.messageGettextItems);
    });

    this.ui.writeLine(chalk.green.bold(`\nExtracted ${files.length} files ✔`));
  },
```

**Comment by mydea** [comment]
This has been fixed in 3.2.0, which now uses a different parser which also allows to parse ts files! I haven't tried this in a production app, but there is a test that covers that. Please let us know if you encounter any issues.
